### ISSUE TITLE
shouldCollapse does not work as intended

### ISSUE THREAD

**Issue by allenlai18** [mac-s-g/react-json-view#371]
i am returning false in many places but it is still showing as collapsed.  I know this is not very descriptive. but i wanted to see if this is an issue others are having as well?

**Comment by tbonfim** [comment]
I just realized `shouldCollapse` callback is only called for field types array and object. I was doing something expecting all values would go through this callback and maybe you are expecting the same thing.

**Comment by timsalesforce** [comment]
Yeah it's not working for me either.  I'm returning false, and the root node is collapsed.  I would like to expand the root node and show the immediate children by default.

**Comment by timsalesforce** [comment]
Ahh, my shouldCollapse() function was async and returned a Promise<boolean>.  I made it synchronous and it works now.
